menas/ui/service/RunService.js

```diff
@@ -126,27 +126,20 @@ var RunService = new function () {
   };
 
   this._buildLineageUrl = function(outputPath, applicationId) {
-    //const urlTemplate = "//localhost:8080/spline-ui/app/lineage-overview?executionEventId=%s"; //TODO unused after #1115 or #1116
-    //const urlTemplate = "//localhost:8080/menas/lineage/app/lineage-overview?executionEventId=%s";
-    //const urlTemplate = "lineage/app/lineage-overview?executionEventId=%s";
-    const urlTemplate = sap.ui.getCore().getModel().getProperty("/lineageService");
-    if (urlTemplate) {
-      const runRestDAO = new RunRestDAO(sap.ui.getCore().getModel().getProperty("/lineageExecutionIdApiTemplate"));
-      const lineageIdInfo = runRestDAO.getLineageId(outputPath, applicationId);
-
-      if (lineageIdInfo.totalCount === 1) {
-        return urlTemplate.replace("%s", lineageIdInfo.executionEventId);
+    const urlTemplate = "lineage/app/lineage-overview?executionEventId=%s";
+    const runRestDAO = new RunRestDAO(sap.ui.getCore().getModel().getProperty("/lineageExecutionIdApiTemplate"));
+    const lineageIdInfo = runRestDAO.getLineageId(outputPath, applicationId);
+
+    if (lineageIdInfo.totalCount === 1) {
+      return urlTemplate.replace("%s", lineageIdInfo.executionEventId);
+    } else {
+      var errorMessage = "";
+      if (!lineageIdInfo.totalCount) {
+        errorMessage = "No lineage found";
       } else {
-        var errorMessage = "";
-        if (!lineageIdInfo.totalCount) {
-          errorMessage = "No lineage found";
-        } else {
-          errorMessage = "Multiple lineage records found";
-        }
-        sap.m.MessageBox.error(errorMessage);
-        return "";
+        errorMessage = "Multiple lineage records found";
       }
-    } else {
+      sap.m.MessageBox.error(errorMessage);
```

Comment:
I have a code suggestion for line 133-144
```javascript
if (lineageIdInfo.totalCount !== 1) {
  let errorMessage = !!lineageIdInfo.totalCount ? "Multiple lineage records found" : "No lineage found";
  sap.m.MessageBox.error(errorMessage);
  return "";
}

return urlTemplate.replace("%s", lineageIdInfo.executionEventId);
```

basically, if something went wrong deal with it, otherwise let the method flow to its end. This, in my opinion, reads more like JS, than again, I am no JS guru.

Reply:
I plan to address this section of the code in #1117 . I might use your suggestion then.